### PR TITLE
sddm: 0.13.0 -> 0.14.0

### PR DESCRIPTION
**Pull request by jokogr**
###### Motivation for this change

To update this package.
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/display-managers/sddm/default.nix [hunk 1]
```diff
@@ -93,6 +88,4 @@ stdenv.mkDerivation {
 
     runHook postInstall
   '';
-
-  inherit (unwrapped) meta;
```
Comment (groxxda) [inline pkgs/applications/display-managers/sddm/default.nix]: Why did you remove this?

**Review by groxxda — CHANGES_REQUESTED** [review]

File: pkgs/applications/display-managers/sddm/default.nix (hunk 1 above)
Comment (jokogr) [inline pkgs/applications/display-managers/sddm/default.nix]: Check https://github.com/jokogr/nixpkgs/blob/5e0f9dacf090191adfd7a1cea6f13adbf6c66f71/pkgs/applications/display-managers/sddm/default.nix#L70

It's already inherited, correctly?

**Comment by jokogr** [comment]
Just a second, I have erased the third patch (although still not in mainline) without checking if it builds with it...

**Comment by jokogr** [comment]
ok, I have also included the "Fix display of user avatars" patch.

File: pkgs/applications/display-managers/sddm/default.nix [hunk 2]
```diff
@@ -69,7 +68,7 @@ in
 stdenv.mkDerivation {
   name = "sddm-${version}";
 
-  nativeBuildInputs = [ lndir makeQtWrapper ];
+  nativeBuildInputs = [ extra-cmake-modules lndir makeQtWrapper ];
```
Comment (abbradar) [inline pkgs/applications/display-managers/sddm/default.nix]: Is this really needed? This derivation is a wrapper so it shouldn't need any CMake-related dependencies...

File: pkgs/applications/display-managers/sddm/default.nix (hunk 1 above)
Comment (abbradar) [inline pkgs/applications/display-managers/sddm/default.nix]: This inherits SDDM's `meta` attribute so that it shows description, maintainer and other meta-information in `nix-env` and related tools.

Comment (jokogr) [inline pkgs/applications/display-managers/sddm/default.nix]: @abbradar @groxxda there was already a `inherit unwrapped;` line, doesn't that mean that the `meta` attribute is already inherited?

File: pkgs/applications/display-managers/sddm/default.nix (hunk 2 above)
Comment (jokogr) [inline pkgs/applications/display-managers/sddm/default.nix]: That seems to be correct, I'm going to remove this.

**Comment by abbradar** [comment]
Offtopic: does anyone also receive two copies of e-mail notifications for each GitHub comment?

File: pkgs/applications/display-managers/sddm/default.nix (hunk 1 above)
Comment (abbradar) [inline pkgs/applications/display-managers/sddm/default.nix]: No, in Nix `{ inherit (foo) bar; }` desugars to `{ bar = foo.bar; }` and `{ inherit baz; }` to `{ baz = baz; }`. So with `inherit unwrapped;` this package would have an `unwrapped` attribute but not a `meta` attribute.

Comment (jokogr) [inline pkgs/applications/display-managers/sddm/default.nix]: @abbradar many thanks for the explanation, I am going to revert it

**Comment by jokogr** [comment]
@abbradar: nope, here it's working as intended

**Comment by jokogr** [comment]
Hopefully all the issues that were mentioned are now properly addressed. Many thanks for your comments!

**Comment by jokogr** [comment]
I had also set the default theme to empty, since maui is now embedded in the greeter.

**Comment by ttuegel** [comment]
Have you run the NixOS tests (`nix-build nixos/release.nix -A tests.sddm`)?

**Comment by jokogr** [comment]
ok, that took a while. It fails, but it keeps building stuff (mesa?) across the multiple runs I've tried. I will have a more careful look later.

**Comment by jokogr** [comment]
@ttuegel I have studied the test and FWIW:

- autologin is working on my machine with xmonad as default window manager.
- with autologin and when icewm is enabled and the default option, it falls back to xmonad.

On nixos-unstable with my PR rebased, when I run the sddm test, I get the following at some point:

```
machine# [   10.403426] systemd[1]: Started User Manager for UID 1000.
machine: running command: test -e /home/alice/.Xauthority
machine# [   10.405355] sddm-helper[892]: Starting: "/nix/store/693bm2nwmmj1hbdazlzhmv4pjl6pglb7-xsession /nix/store/693bm2nwmmj1hbdazlzhmv4pjl6pglb7-xsession 'none + icewm'"
machine: exit status 0
machine: must succeed: xauth merge ~alice/.Xauthority
machine# [   10.408992] sddm-helper[904]: Adding cookie to "/home/alice/.Xauthority"
machine# xauth:  file /root/.Xauthority does not exist
machine: exit status 0
machine: waiting for a window to appear
machine: must succeed: xwininfo -root -tree | sed 's/.*0x[0-9a-f]* \"\([^\"]*\)\".*/\1/; t; d'
machine# [   10.447779] sddm[877]: Session started
machine: exit status 0
machine# [   11.351133] xsession[904]: IceWM: using /home/alice/.icewm for private configuration files
machine: must succeed: xwininfo -root -tree | sed 's/.*0x[0-9a-f]* \"\([^\"]*\)\".*/\1/; t; d'
machine: exit status 0
collecting coverage data
machine: running command: test -e /sys/kernel/debug/gcov
machine: exit status 1
syncing
machine: running command: sync
machine: exit status 0
cleaning up
```

**Comment by ttuegel** [comment]
Evidently that test has been broken for a few days. I'll investigate.

**Comment by ttuegel** [comment]
I've corrected the failing test now, so this is good to go.
